### PR TITLE
Add support for pacman package version query

### PR DESCRIPTION
**Pull request by xatier**
- refactor package query logic to be package manager agnostic.

- `pacman -Qo` is the equivalent to `rpm -qf` [1].

[1] https://wiki.archlinux.org/index.php/Pacman/Rosetta#Querying_specific_packages

----

Changes are tested on my local box (Arch) and a Ubuntu 20.04 container. See below for details.

## tested on Arch

```
$ make binaries
go build -mod=vendor  -gcflags 'all=-trimpath=/home/xatier/tmp/podman' -asmflags 'all=-trimpath=/home/xatier/tmp/podman' -ldflags '-X github.com/containers/podman/v2/libpod/define.gitCommit=e4e9a139024c771c70a429e9d818fdfe3e63dcbe-dirty -X github.com/containers/podman/v2/libpod/define.buildInfo=1608062677 -X github.com/containers/podman/v2/libpod/config._installPrefix=/usr/local -X github.com/containers/podman/v2/libpod/config._etcDir=/etc ' -tags "    systemd exclude_graphdriver_devicemapper seccomp" -o bin/podman ./cmd/podman
go build -mod=vendor  -gcflags 'all=-trimpath=/home/xatier/tmp/podman' -asmflags 'all=-trimpath=/home/xatier/tmp/podman' -ldflags '-X github.com/containers/podman/v2/libpod/define.gitCommit=e4e9a139024c771c70a429e9d818fdfe3e63dcbe-dirty -X github.com/containers/podman/v2/libpod/define.buildInfo=1608062680 -X github.com/containers/podman/v2/libpod/config._installPrefix=/usr/local -X github.com/containers/podman/v2/libpod/config._etcDir=/etc ' -tags "remote exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper containers_image_openpgp" -o bin/podman-remote ./cmd/podman

$ ./bin/podman info                                                        
host:                                                                                                    
  arch: amd64        
  buildahVersion: 1.19.0-dev   
  cgroupManager: cgroupfs                                                                                
  cgroupVersion: v1                                                                                      
  conmon:                                           
    package: /usr/bin/conmon is owned by conmon 1:2.0.21-1   <------------ pacman -Qo conmon
    path: /usr/bin/conmon
    version: 'conmon version 2.0.21, commit: 35a2fa83022e56e18af7e6a865ba5d7165fa2a4a'
  cpus: 16
  distribution:
    distribution: arch
    version: unknown
  eventLogger: journald
  hostname: walla
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.8.14-arch1-1
  linkmode: dynamic
  memFree: 20187516928
  memTotal: 135012790272
  ociRuntime:
    name: crun
    package: /usr/bin/crun is owned by crun 0.16-1   <------------ pacman -Qo crun
    path: /usr/bin/crun
    version: |-
      crun version 0.16
      commit: eb0145e5ad4d8207e84a327248af76663d4e50dd
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
...
```

## tested on Ubuntu (container)

```
root@8b9f24f29ee1:/podman# make binaries
go build -mod=vendor  -gcflags 'all=-trimpath=/podman' -asmflags 'all=-trimpath=/podman' -ldflags '-X github.com/containers/podman/v2/libpod/define.gitCommit=e4e9a139024c771c70a429e9d818fdfe3e63dcbe -X github.com/containers/podman/v2/libpod/define.buildInfo=1608062788 -X github.com/containers/podman/v2/libpod/config._installPrefix=/usr/local -X github.com/containers/podman/v2/libpod/config._etcDir=/etc ' -tags "   selinux systemd exclude_graphdriver_devicemapper seccomp" -o bin/podman ./cmd/podman
go build -mod=vendor  -gcflags 'all=-trimpath=/podman' -asmflags 'all=-trimpath=/podman' -ldflags '-X github.com/containers/podman/v2/libpod/define.gitCommit=e4e9a139024c771c70a429e9d818fdfe3e63dcbe -X github.com/containers/podman/v2/libpod/define.buildInfo=1608062793 -X github.com/containers/podman/v2/libpod/config._installPrefix=/usr/local -X github.com/containers/podman/v2/libpod/config._etcDir=/etc ' -tags "remote exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper containers_image_openpgp" -o bin/podman-remote ./cmd/podman

root@8b9f24f29ee1:/podman# ./bin/podman info 
host:
  arch: amd64
  buildahVersion: 1.19.0-dev
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: 'conmon: /usr/libexec/podman/conmon'  <------- same behavior as `dpkg -S /usr/bin/conmon`
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.20, commit: '
  cpus: 16
  distribution:
    distribution: ubuntu
    version: "20.04"
  eventLogger: journald
  hostname: 8b9f24f29ee1
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.8.14-arch1-1
  linkmode: dynamic
  memFree: 20158173184
  memTotal: 135012790272
  ociRuntime:
    name: crun
    package: 'crun: /usr/bin/crun'  <------- same behavior as `dpkg -S /usr/bin/crun`
    path: /usr/bin/crun
    version: |-
      crun version 0.16.3-fd58-dirty
      commit: fd582c529489c0738e7039cbc036781d1d039014
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
```

### CODE REVIEW

**Comment by rhatdan** [comment]
Please sign your commits.
git commit -a -s --amend
git push --force

LGTM
/approve
Thanks @xatier for the PR.

File: libpod/util.go [hunk 1]
```diff
@@ -142,29 +142,35 @@ func JSONDeepCopy(from, to interface{}) error {
 	return json.Unmarshal(tmp, to)
 }
 
-func dpkgVersion(path string) string {
+func queryPackageVersion(cmdArg ...string) string {
 	output := unknownPackage
-	cmd := exec.Command("/usr/bin/dpkg", "-S", path)
+	cmd := exec.Command(cmdArg[0], cmdArg[1:]...)
```
Comment (TomSweeneyRedHat) [inline libpod/util.go:147]: I know the code is all fine, but just to make sure in the future, you probably ought to add a len check for cmdArg before this line.

Comment (xatier) [reply libpod/util.go:147]: Hey @TomSweeneyRedHat , thanks for pointing this out.

How do you think about this? I'm not sure if we should have a max len for `cmdArg` though.

```go
func queryPackageVersion(cmdArg ...string) string {                             
    output := unknownPackage
    if 1 < len(cmdArg) {                                                    
        cmd := exec.Command(cmdArg[0], cmdArg[1:]...)                               
        if outp, err := cmd.Output(); err == nil {                                  
            output = string(outp)                                                   
        }                           
    }                                                
    return strings.Trim(output, "\n")                                           
}
```

Comment (rhatdan) [reply libpod/util.go:147]: LGTM, no max check.

Comment (xatier) [reply libpod/util.go]: @rhatdan, awesome, I've amended my PR. Please let me know if you have any other suggestions. :+1:

**Comment by zhangguanzhang** [comment]
title iswrong，should be podman

**Comment by rhatdan** [comment]
@xatier Could you rebase and repush, Should clear up the failed test, and we can get this merged.

**Comment by xatier** [comment]
sure, I've rebased with containers/podman's master.

**Comment by rhatdan** [comment]
/lgtm
/hold
Thanks @xatier

**Pull request merged**